### PR TITLE
[webui] Fix broken flags routes

### LINKED ISSUE #2007: Changing flags in webui is not working anymore
When changing a flag the webui redirects to an unknown URL.  Tested to disable global and architecture specfic flag with current git master.

It is also possible to click on the "Disable" popup when not logged-in which creates a not nice "nilClass" error notify.

Not routed URLs are for example:

https://build-test.opensuse.org/flag/OBS:Server:Unstable/obs-service-tar_scm?architecture=x86_64&flag=build&status=disable

https://build-test.opensuse.org/flag/OBS:Server:Unstable/obs-service-tar_scm?architecture=&flag=build&status=disable

### PR DESCRIPTION
**Pull request by ChrisBr**
flag wasn't found in the controller because of broken package route.
fixes #2007.

### CODE REVIEW

File: src/api/config/routes.rb [hunk 1]
```diff
@@ -174,7 +174,7 @@ def self.matches?(request)
       post 'project/create_image_repository' => :create_image_repository
 
       # Flags
-      post 'flag/:project(/:package)/:flag' => :toggle_flag, constraints: cons
```
Comment (bgeuken) [inline src/api/config/routes.rb]: Do you know why this did not work on the package tab, but for projects?

**Comment by bgeuken** [comment]
Tested for project and package flags and it seems to work for both now.

One thing I noticed is that when a user doesn't have permissions to change the flags, he won't see the error notification. And the spinner icon won't disappear. We should fix that as well.

**Comment by ChrisBr** [comment]
@bgeuken I will fix this in a separate PR

File: src/api/config/routes.rb (hunk 1 above)
Comment (ChrisBr) [inline src/api/config/routes.rb]: Seems like the URL helper produced wrong URLs and we ended up with

```
'flag/home:Admin/testpackage?flag=42
```
and then the controller interpreted ``testpackage`` as the flag id and tried to fetch ``Flag.find('testpackage')`` which obviously failed. In Project the URL was correct. But anyway, I think using `PUT` for `:update` and `POST` for `:create` and dropping the flag from the route is anyway more the rails way...

**Comment by bgeuken** [comment]
Fine with me

**Pull request merged**